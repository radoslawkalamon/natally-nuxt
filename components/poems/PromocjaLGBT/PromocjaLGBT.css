.transition-lgbt-image-wrapper-enter-active,
.transition-lgbt-image-wrapper-leave-active {
  transition: opacity 0.5s linear;
}

.transition-lgbt-image-wrapper-enter,
.transition-lgbt-image-wrapper-leave-to {
  opacity: 0;
}

@keyframes animation-lgbt-reaction-wrapper-heart {
  0% {
    top: -64px;
  }

  100% {
    top: calc(100% + 64px);
  }
}

.lgbt-product-card {
  --product-card-image-height: 200px;
  --product-card-image-width: 280px;

  border: var(--border-size) solid var(--color-floating-border);
  border-radius: var(--border-radius);
  box-shadow: 0 8px 16px var(--color-floating-shadow);
  cursor: pointer;
  font-family: var(--font-family-sans);
  font-weight: 700;
  margin: 0 auto;
  width: var(--product-card-image-width);
}

.lgbt-product-card__image-wrapper {
  height: var(--product-card-image-height);
  position: relative;
}

.lgbt-product-card__image {
  border-radius: var(--border-radius) var(--border-radius) 0 0;
  display: block;
  height: var(--product-card-image-height);
  left: 0;
  position: absolute;
  top: 0;
  width: var(--product-card-image-width);
  z-index: 2;
}

.lgbt-product-card__content-wrapper {
  align-items: center;
  display: flex;
  flex-flow: column nowrap;
  gap: 24px;
  padding: 16px;
  position: relative;
  text-align: center;
}

.lgbt-product-card__ribbon {
  align-items: center;
  background-color: var(--color-off-white-1);
  border: 2px solid var(--color-accent);
  border-radius: 50%;
  display: flex;
  height: 48px;
  justify-content: center;
  left: 16px;
  position: absolute;
  top: 0;
  transform: translateY(-50%);
  width: 48px;
  z-index: 3;
}

.lgbt-product-card__ribbon-icon {
  height: calc(100% / 1.5);
  width: calc(100% / 1.5);
}

.lgbt-product-card__title {
  margin: 0 !important;
}

.lgbt-product-card__price {
  display: inline-block;
  font-size: 1.25em;
  position: relative;
}

.lgbt-product-card__price::after {
  background-color: var(--color-accent);
  border-radius: 100px;
  bottom: 75%;
  color: var(--color-off-white-1);
  content: '-100%';
  display: block;
  font-size: 0.6em;
  left: calc(100% + 4px);
  line-height: 1;
  padding: 0.25em 0.5em;
  position: absolute;
}

.lgbt-reaction-wrapper__heart {
  animation-iteration-count: 1;
  animation-name: animation-lgbt-reaction-wrapper-heart;
  animation-timing-function: ease-in;
  height: 64px;
  left: 0;
  position: fixed;
  top: -64px;
  transform: translateX(-50%);
  width: 64px;
  z-index: 9;
}

.lgbt-product-card__add-to-cart {
  align-items: center;
  appearance: none;
  background-color: var(--color-accent);
  border: 0;
  border-radius: var(--border-radius);
  color: var(--color-off-white-1);
  cursor: pointer;
  display: flex;
  font-family: var(--font-family-sans);
  font-size: 0.75em;
  font-weight: bold;
  gap: 0.5em;
  margin: 0;
  padding: 0.5em 1em;
}

.lgbt-product-card__add-to-cart:hover {
  text-decoration: underline;
}

.lgbt-product-card__add-to-cart-icon {
  height: 24px;
  width: 24px;
}
