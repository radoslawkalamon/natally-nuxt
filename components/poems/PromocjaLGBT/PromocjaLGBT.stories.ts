import type { Meta, Story } from '@storybook/vue'
import PoemsPromocjaLGBT from '@/components/poems/PromocjaLGBT/PromocjaLGBT.vue'
import WrappersTextPoems from '@/components/wrappers/TextPoems/TextPoems.vue'

const meta: Meta = {
  title: 'Poems / Promocja LGBT',
  component: PoemsPromocjaLGBT
}
export default meta

export const Default: Story = () => ({
  components: {
    PoemsPromocjaLGBT,
    WrappersTextPoems
  },
  template: `<div class="scoped__storybook-section">
    <WrappersTextPoems>
      <PoemsPromocjaLGBT />
    </WrappersTextPoems>
  </div>`
})
