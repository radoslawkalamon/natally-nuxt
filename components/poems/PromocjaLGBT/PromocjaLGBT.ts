import throttle from 'lodash/throttle'
import Vue from 'vue'

type PoemsPromocjaLGBTData = {
  activeImage: number;
  startAnimation: boolean;
  sliderInterval: number | NodeJS.Timeout;
}

const animationDuration = 2000

export default Vue.extend({
  name: 'PoemsPromocjaLGBT',
  data (): PoemsPromocjaLGBTData {
    return {
      activeImage: 0,
      startAnimation: false,
      sliderInterval: 0
    }
  },
  computed: {
    images (): Array<Record<string, string>> {
      const images = [
        {
          alt: 'Dwie uśmiechnięte kobiety w łazience. Jedna z nich wyciera ręcznikiem drugą.',
          '1x': require('./images/promocja-lgbt-1.1x.webp'),
          '2x': require('./images/promocja-lgbt-1.2x.webp')
        },
        {
          alt: 'Dwoje przytulających się, śmiejących się mężczyzn.',
          '1x': require('./images/promocja-lgbt-2.1x.webp'),
          '2x': require('./images/promocja-lgbt-2.2x.webp')
        },
        {
          alt: 'Dwie kobiety, ubrane w suknie ślubne. Jedna całuje czoło drugiej.',
          '1x': require('./images/promocja-lgbt-3.1x.webp'),
          '2x': require('./images/promocja-lgbt-3.2x.webp')
        },
        {
          alt: 'Dwoje mężczyzn, siedzących na trawie, śmiejących się i przytulających się',
          '1x': require('./images/promocja-lgbt-4.1x.webp'),
          '2x': require('./images/promocja-lgbt-4.2x.webp')
        }
      ]
      return images.map(image => ({
        alt: image.alt,
        srcset: `${image['1x']} 1x, ${image['2x']} 2x`,
        src: image['1x']
      }))
    },
    imagesCount (): number {
      return this.images.length
    },
    heartCount (): number {
      return 69
    },
    heartStyles (): Array<Record<string, string>> {
      return Array(this.heartCount).fill(null).map(_i => ({
        left: `${this.getRandomNumberBetween(0, 100)}%`,
        opacity: this.getRandomNumberBetween(0.6, 1),
        animationDuration: `${this.getRandomNumberBetween(1, animationDuration / 1000)}s`
      }))
    }
  },
  created (): void {
    this.buyClick = throttle(this.buyClick, animationDuration, { trailing: false })

    this.sliderInterval = setInterval(() => this.changeSlide(), 3000)
  },
  destroyed (): void {
    clearInterval(Number(this.sliderInterval))
  },
  methods: {
    changeSlide (): void {
      this.activeImage = (this.activeImage + 1) % this.imagesCount
    },
    getRandomNumberBetween (min: number, max: number): string {
      return ((Math.random() * (max - min)) + min).toString()
    },
    buyClick (): void {
      this.startAnimation = true
      setTimeout(() => (this.startAnimation = false), animationDuration)
    }
  }
})
