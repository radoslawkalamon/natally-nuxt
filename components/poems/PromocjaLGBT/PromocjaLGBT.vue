<template>
  <div class="promocja-lgbt">
    <div
      class="lgbt-product-card"
      @click="buyClick"
    >
      <transition-group
        class="lgbt-product-card__image-wrapper"
        name="transition-lgbt-image-wrapper"
        tag="div"
      >
        <template v-for="(image, index) in images">
          <img
            v-if="index === activeImage"
            :key="index"
            class="lgbt-product-card__image"
            v-bind="image"
          >
        </template>
      </transition-group>
      <div class="lgbt-product-card__content-wrapper">
        <div class="lgbt-product-card__ribbon">
          <img
            alt="Produkt w promocji!"
            class="lgbt-product-card__ribbon-icon"
            src="./images/ribbon.svg"
            title="Produkt w promocji!"
          >
        </div>
        <h2
          class="lgbt-product-card__title"
          v-text="labels.productTitle"
        />
        <span
          class="lgbt-product-card__price"
          v-text="labels.productPrice"
        />
        <button
          aria-label="Dodaj do koszyka"
          class="lgbt-product-card__add-to-cart"
          data-test="promocja-lgbt-buy-now-button"
          title="Dodaj do koszyka"
          type="button"
        >
          <img
            class="lgbt-product-card__add-to-cart-icon"
            src="./images/magnifier.svg"
          >
          <span v-text="labels.productBuyButton" />
        </button>
      </div>
      <div
        v-if="startAnimation"
        class="lgbt-reaction-wrapper"
      >
        <template v-for="i in heartCount">
          <img
            :key="i"
            alt="Serce"
            class="lgbt-reaction-wrapper__heart"
            data-test="promocja-lgbt-heart"
            src="./images/heart.svg"
            :style="heartStyles[i]"
            title="Serce"
          >
        </template>
      </div>
    </div>
  </div>
</template>

<script src="./PromocjaLGBT.ts" lang="ts"></script>
<style src="./PromocjaLGBT.css" scoped></style>
