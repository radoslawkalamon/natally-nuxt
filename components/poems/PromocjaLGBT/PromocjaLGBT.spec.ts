import PromocjaLGBT from './PromocjaLGBT.vue'
import { shallRender } from '@/devtools/jest.shared.spec'

describe('Poems / Promocja LGBT', () => {
  shallRender(PromocjaLGBT)
})
