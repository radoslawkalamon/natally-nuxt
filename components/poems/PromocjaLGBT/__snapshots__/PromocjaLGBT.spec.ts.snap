// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poems / Promocja LGBT Unit shall change slide after 3 seconds 1`] = `
"<div class="promocja-lgbt">
  <div class="lgbt-product-card">
    <transition-group-stub tag="div" name="transition-lgbt-image-wrapper" class="lgbt-product-card__image-wrapper">
      <!----><img alt="Dwoje przytulających się, śmiejących się mężczyzn." srcset="promocja-lgbt-2.1x.webp 1x, promocja-lgbt-2.2x.webp 2x" src="promocja-lgbt-2.1x.webp" class="lgbt-product-card__image">
      <!---->
      <!---->
    </transition-group-stub>
    <div class="lgbt-product-card__content-wrapper">
      <div class="lgbt-product-card__ribbon"><img alt="Produkt w promocji!" src="./images/ribbon.svg" title="Produkt w promocji!" class="lgbt-product-card__ribbon-icon"></div>
      <h2 class="lgbt-product-card__title">Miłość</h2> <span class="lgbt-product-card__price">0,00 zł</span> <button aria-label="Dodaj do koszyka" data-test="promocja-lgbt-buy-now-button" title="Dodaj do koszyka" type="button" class="lgbt-product-card__add-to-cart"><img src="./images/magnifier.svg" class="lgbt-product-card__add-to-cart-icon"> <span>Sprawdź!</span></button>
    </div>
    <!---->
  </div>
</div>"
`;

exports[`Poems / Promocja LGBT Unit shall pass unit sanity test 1`] = `
"<div class="promocja-lgbt">
  <div class="lgbt-product-card">
    <transition-group-stub tag="div" name="transition-lgbt-image-wrapper" class="lgbt-product-card__image-wrapper"><img alt="Dwie uśmiechnięte kobiety w łazience. Jedna z nich wyciera ręcznikiem drugą." srcset="promocja-lgbt-1.1x.webp 1x, promocja-lgbt-1.2x.webp 2x" src="promocja-lgbt-1.1x.webp" class="lgbt-product-card__image">
      <!---->
      <!---->
      <!---->
    </transition-group-stub>
    <div class="lgbt-product-card__content-wrapper">
      <div class="lgbt-product-card__ribbon"><img alt="Produkt w promocji!" src="./images/ribbon.svg" title="Produkt w promocji!" class="lgbt-product-card__ribbon-icon"></div>
      <h2 class="lgbt-product-card__title">Miłość</h2> <span class="lgbt-product-card__price">0,00 zł</span> <button aria-label="Dodaj do koszyka" data-test="promocja-lgbt-buy-now-button" title="Dodaj do koszyka" type="button" class="lgbt-product-card__add-to-cart"><img src="./images/magnifier.svg" class="lgbt-product-card__add-to-cart-icon"> <span>Sprawdź!</span></button>
    </div>
    <!---->
  </div>
</div>"
`;
